src/components/Form.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { FormEvent } from 'react';
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+interface State {
+  userId: number;
+  title: string;
+  isUserSelected: boolean,
+  isTitleEntered: boolean,
+}
+
+interface Props {
+  todos: Todo[];
+  users: User[];
+  addTodo: (todo: Todo) => void;
+}
+
+export class Form extends React.Component<Props, State> {
+  state = {
+    userId: 0,
+    title: '',
+    isUserSelected: true,
+    isTitleEntered: true,
+  };
+
+  handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const { userId, title } = this.state;
+    const { users, todos, addTodo } = this.props;
+
+    if (userId === 0) {
+      this.setState({ isUserSelected: false });
+
+      return;
+    }
+
+    if (title.length === 0) {
+      this.setState({ isTitleEntered: false });
+
+      return;
+    }
+
+    this.setState(() => {
+      const newTodo = {
+        user: users.find(user => user.id === userId) || null,
+        userId,
+        id: todos.length + 1,
```

Comment:
what if some todos will be deleted? then the length will be decreased and id will be repeated